rclcpp/include/rclcpp/executor.hpp

```diff
@@ -215,6 +217,26 @@ class Executor
     }
   }
 
+  /* The default implementation of Executor::get_allocated_handles dynamically
+     allocates the requested handle array, and ignores the parent pointer. */
+  virtual void ** get_allocated_handles(executor_handle_t /*handle_type*/,
```

Comment:
Maybe these functions should just be `borrow_handles` and `return_handles`? Or maybe `get_handles` and `return_handles`? It seems like using the term allocate here may not always be true.

Also, I guess you're commenting out the parameter name to avoid an unused parameter warning?